app/frontend/src/features/profile/view/leaveReference/formSteps/submit/SubmitReference.tsx

```diff
@@ -0,0 +1,134 @@
+  referenceData: ReferenceContextFormData;
+}
+
+export default function SubmitReference({
+  user,
+  referenceData,
+}: SubmitReferenceProps) {
+  const {
+    writeFriendReference,
+    status: friendReferenceWritingStatus,
+    reset: resetFriendReferenceWriting,
+  } = useWriteFriendReference(user.userId);
+  const {
+    writeHostRequestReference,
+    status: hostRequestReferenceWritingStatus,
+    reset: resetHostRequestReferenceWriting,
+  } = useWriteHostReference(user.userId);
+
+  const { referenceType, hostRequest } = useParams<{
+    referenceType: string;
+    hostRequest?: string;
+  }>();
+  const theme = useTheme();
+  const classes = useReferenceStyles();
+  const isSmOrWider = useMediaQuery(theme.breakpoints.up("sm"));
+  const { handleSubmit } = useForm<ReferenceContextFormData>();
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+
+  const onFriendReferenceSubmit = () => {
+    const formData: WriteFriendReferenceInput =
+      referenceData.wasAppropriate === "true"
+        ? {
+            toUserId: user.userId,
+            wasAppropriate: true,
+            text: referenceData.text,
+            rating: referenceData.rating,
+          }
+        : {
+            toUserId: user.userId,
+            wasAppropriate: false,
+            text: referenceData.text,
+            rating: referenceData.rating,
+          };
+
+    resetFriendReferenceWriting();
+    writeFriendReference({
+      referenceData: formData,
+    });
+    window.scroll({ top: 0 });
+  };
+
+  const onHostReferenceSubmit = () => {
+    if (hostRequest && +hostRequest !== undefined) {
+      const formData: WriteHostRequestReferenceInput =
+        referenceData.wasAppropriate === "true"
+          ? {
+              hostRequestId: +hostRequest,
+              wasAppropriate: true,
+              text: referenceData.text,
+              rating: referenceData.rating,
+            }
+          : {
+              hostRequestId: +hostRequest,
+              wasAppropriate: false,
+              text: referenceData.text,
+              rating: referenceData.rating,
+            };
+
+      resetHostRequestReferenceWriting();
+      writeHostRequestReference({
+        referenceData: formData,
+      });
+      window.scroll({ top: 0 });
+    } else {
+      setErrorMessage("Error fetching the host request.");
+    }
+  };
+
+  let onSubmit = onHostReferenceSubmit;
+  if (referenceType === referenceTypeRoute[0]) {
+    onSubmit = onFriendReferenceSubmit;
+  }
+
+  return (
+    <>
+      {(friendReferenceWritingStatus || hostRequestReferenceWritingStatus) ===
+      "success" ? (
+        <Alert className={classes.alert} severity="success">
+          {REFERENCE_SUCCESS}
+        </Alert>
+      ) : (friendReferenceWritingStatus ||
+          hostRequestReferenceWritingStatus) === "error" ||
+        errorMessage !== null ? (
+        <Alert className={classes.alert} severity="error">
+          {errorMessage || "Unknown error"}
```

Comment:
Use `error` from the mutations, you'll get a better server message. If you still need "Unknown error", move it to constants